presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/NodeSet.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.teradata.prestomanager.controller;
+
+import com.teradata.prestomanager.controller.NodeSet.Agent.NodeRole;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+import static com.teradata.prestomanager.controller.NodeSet.Agent.NodeRole.COORDINATOR;
+import static com.teradata.prestomanager.controller.NodeSet.Agent.NodeRole.WORKER;
+import static java.util.Objects.requireNonNull;
+
+@ThreadSafe
+public class NodeSet
+{
+    private static NodeSet nodeSetInstance = new NodeSet();
+
+    public static Set<Agent> nodeSet = ConcurrentHashMap.newKeySet();
+
+    public static NodeSet getInstance()
+    {
+        return nodeSetInstance;
+    }
+
+    public void addAgent(URI uri, NodeRole nodeRole)
+    {
+        nodeSet.add(new Agent(uri, nodeRole));
+    }
+
+    public void removeAgent(int id)
+    {
+        nodeSet.removeIf(agent -> agent.getId().equals(id));
+    }
+
+    public Integer getIdByUri(URI uri)
+    {
+        return nodeSet.stream().filter(agent -> agent.getUri().equals(uri))
+                .map(Agent::getId).findAny().orElseThrow(() -> new IllegalArgumentException("Invalid URI"));
+    }
+
+    public Collection<URI> getUrisByIds(Collection<Integer> ids)
+    {
+        Collection<URI> uriList = nodeSet.stream()
+                .filter(agent -> ids.contains(agent.getId()))
+                .map(Agent::getUri).collect(Collectors.toList());
+        if (ids.size() != uriList.size()) {
+            throw new IllegalArgumentException("Invalid node ID");
```

Comment:
This means one of the ```ids``` is invalid. Modify the error message and also add it to logger.